src/battle_ai_util.c

```diff
@@ -833,6 +834,16 @@ s32 AI_CalcDamage(u16 move, u8 battlerAtk, u8 battlerDef, u8 *typeEffectiveness,
             case EFFECT_FINAL_GAMBIT:
                 dmg = gBattleMons[battlerAtk].hp;
                 break;
+            #if B_BEAT_UP >= GEN_5
+            case EFFECT_BEAT_UP:
+                gBattleStruct->beatUpSlot = 0;
+                dmg = 0;
+                for (i = 0; i < CalculatePartyCount(GetBattlerParty(battlerAtk)); i++) {
```

Comment:
Wouldn't it be better to move CalculatePartyCount out of the loop to avoid running the function more than one time?

Reply:
Ah I assumed the compiler would be smart enough to pre compute it. Will do though

Reply:
The compiler doesn't know what CalculatePartyCount does, so it has to call it every time.